python_modules/dagster/dagster/_core/definitions/partitions/definition/time_window.py

```diff
@@ -194,6 +194,17 @@ def __new__(
             exclusions=cleaned_exclusions if cleaned_exclusions else None,
         )
 
+    def validate_partition_definition(self) -> None:
+        # Try to determine if there are multiple time ranges being mapped
+        # to the same partition key.
+        first_partition_key = self.get_first_partition_key()
+        if self.get_next_partition_key(first_partition_key) == first_partition_key:
+            raise DagsterInvalidDefinitionError(
+                "This partition set contains multiple time ranges that map to the same partition key. "
+                f"This usually indicates that the partition set's format string ({self.fmt}) is "
+                f"not granular enough to produce a unique key for each time in the cron schedule ({self.cron_schedule})."
+            )
```

Comment:
somewhat concerned about cases where you have a time window partitions def that contains no or 1 partition. (e.g. start date is a future date or yesterday). Can you add some tests for those cases?